sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SchemaPruning.scala

```diff
@@ -21,9 +21,12 @@ import org.apache.spark.sql.types._
 
 object SchemaPruning {
   /**
-   * Filters the schema by the requested fields. For example, if the schema is struct<a:int, b:int>,
-   * and given requested field are "a", the field "b" is pruned in the returned schema.
-   * Note that schema field ordering at original schema is still preserved in pruned schema.
+   * Prunes the nested schema by the requested fields. For example, if the schema is
+   * struct<a:int, b:int>, and given requested field are "a", the field "b" is pruned in the
+   * returned schema.
+   * Note that:
+   *   1. The schema field ordering at original schema is still preserved in pruned schema.
+   *   2. The top-level fields are not pruned here.
```

Comment:
Hmm, doesn't it mean we miss the change to prune top-level columns?

Reply:
No. Top-level columns pruned by:
https://github.com/apache/spark/blob/7a5647a93aaea9d1d78d9262e24fc8c010db04d0/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala#L208-L213

Reply:
I think v2 column pruning should be handled by `V2ScanRelationPushDown`.

Reply:
Yes, I have updated the v2 part:
https://github.com/apache/spark/blob/a966bac379ff6fed20c57cf3c748cada9da28b4c/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/PushDownUtils.scala#L79-L110